### PR TITLE
Atome - can't create source object

### PR DESCRIPTION
**Pull request by vault087**
Due specific requirements to PaymentInformation for Atome it required refactoring on API level. ViewController wasn't implemented good enough to support these changes so it was recreated from scratch.

Some tools were added to support unit testing and live UI development.

### CODE REVIEW

File: OmiseSDK/AtomeFormViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                  street2: viewContext[.street2])
+
+        let items: [PaymentInformation.Atome.Item] = [
+            .init(
+                sku: "3427842",
+                category: "Shoes",
+                name: "Prada shoes",
+                quantity: 1,
+                amount: flowSession.paymentAmount ?? 0,
+                itemUri: "www.kan.com/product/shoes",
```
Comment (danfowler) [inline OmiseSDK/AtomeFormViewModel.swift:86]: @vault087 we should avoid adding unowned domains into our source code. Better to use "omise.co/product/shoes/" or something.

File: OmiseSDK/Helpers/Bundle+OmiseSDK.swift [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import Foundation
+
+private final class OmiseSDK216d048c5feca16469a7a71566256537 {
```
Comment (danfowler) [inline OmiseSDK/Helpers/Bundle+OmiseSDK.swift:3]: Is it worth adding a comment explaining why the class is named like this?

File: OmiseSDK/PaymentChooserViewController.swift [hunk 3]
```diff
@@ -417,6 +421,19 @@ class PaymentChooserViewController: AdaptableStaticTableViewController<PaymentCh
...
     }
 
+    func goToAtome() {
+        let vc = AtomeFormViewController(viewModel: AtomeFormViewModel(flowSession: flowSession))
+        vc.preferredPrimaryColor = self.preferredPrimaryColor
+        vc.preferredSecondaryColor = self.preferredSecondaryColor
+
+        navigationController?.pushViewController(vc, animated: true)
+//
+//        if let storyboard = self.storyboard {
```
Comment (danfowler) [inline OmiseSDK/PaymentChooserViewController.swift:431]: Why commented code?

File: OmiseSDK/PaymentInformation.Atome.swift [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            self.phoneNumber = phoneNumber
+            self.shippingAddress = shippingAddress
+            self.items = items
+        }
+    }
+}
+
+public extension PaymentInformation.Atome {
+    struct Item: Codable, Equatable {
+        //        public let sku: String
```
Comment (danfowler) [inline OmiseSDK/PaymentInformation.Atome.swift:56]: Why commented?

File: OmiseSDK/AtomeFormViewModel.swift (hunk 1 above)
Comment (danfowler) [reply OmiseSDK/AtomeFormViewModel.swift:86]: See @swathi1102 for details as to why.

Comment (vault087) [reply OmiseSDK/AtomeFormViewModel.swift:86]: I took the code from Android side, I could change it to any value but would be good if we use the same

File: OmiseSDK/Helpers/Bundle+OmiseSDK.swift (hunk 2 above)
Comment (vault087) [reply OmiseSDK/Helpers/Bundle+OmiseSDK.swift:3]: Don't think so, the next line of code use it and for iOS developer its pretty straight forward

File: OmiseSDK/PaymentInformation.Atome.swift (hunk 4 above)
Comment (vault087) [reply OmiseSDK/PaymentInformation.Atome.swift]: Fixed

File: OmiseSDK/PaymentChooserViewController.swift (hunk 3 above)
Comment (vault087) [reply OmiseSDK/PaymentChooserViewController.swift]: fixed

**Review by danfowler — APPROVED** [review]

**Review by aashishgurung — APPROVED** [review]

**Pull request merged**